src/visual.ts

```diff
@@ -260,122 +255,144 @@ export class Visual implements IVisual {
             return null;
         }
 
-        let columns: DataRoles<DataViewCategoryColumn | DataViewValueColumn> = <any>mapValues(
+        const columns: DataRoles<DataViewCategoryColumn | DataViewValueColumn> = <any>mapValues(
             Visual.RoleDisplayNames,
             (x, i) => Visual.getCategoricalColumnOfRole(dataView, i)
         );
 
-        let valuesColumn: DataViewValueColumn = <DataViewValueColumn>columns.Value;
-        let timeStampColumn = <DataViewCategoryColumn>columns.Timestamp;
+        const valuesColumn: DataViewValueColumn = <DataViewValueColumn>columns.Value;
+        const timeStampColumn = <DataViewCategoryColumn>columns.Timestamp;
 
         if (!timeStampColumn) {
             return null;
         }
 
-        let isScalar: boolean = !(timeStampColumn.source && timeStampColumn.source.type && timeStampColumn.source.type.dateTime);
-
-        const settings: PulseChartSettings = Visual.parseSettings(dataView, colorHelper);
-
-        let categoryValues: any[] = timeStampColumn.values;
+        const categoryValues: any[] = timeStampColumn.values;
 
         if (!categoryValues || isEmpty(dataView.categorical.values) || !valuesColumn || isEmpty(valuesColumn.values)) {
             return null;
         }
-        let minValuesValue = Math.min.apply(null, valuesColumn.values), maxValuesValue = Math.max.apply(null, valuesColumn.values);
-        let minCategoryValue = Math.min.apply(null, categoryValues), maxCategoryValue = Math.max.apply(null, categoryValues);
-        settings.xAxis.dateFormat =
-            (maxCategoryValue - minCategoryValue < (24 * 60 * 60 * 1000)
-                && new Date(maxCategoryValue).getDate() === new Date(minCategoryValue).getDate())
-                ? XAxisDateFormat.TimeOnly
-                : XAxisDateFormat.DateOnly;
 
-        settings.xAxis.formatterOptions = {
-            value: isScalar ? minCategoryValue : new Date(minCategoryValue),
-            value2: isScalar ? maxCategoryValue : new Date(maxCategoryValue)
-        };
-        settings.yAxis.formatterOptions = {
-            value: minValuesValue,
-            value2: maxValuesValue,
-            format: valueFormatter.getFormatString(valuesColumn.source, null)
-        };
+        const minCategoryValue = Math.min.apply(null, categoryValues);
+        const maxCategoryValue = Math.max.apply(null, categoryValues);
+        const isScalar: boolean = Visual.setAxisFormatter(valuesColumn, timeStampColumn, settings, maxCategoryValue, minCategoryValue);
```

Comment:
This is an old way of getting minimum out of array. Try `Math.min(0, ...categoryValues)`

Reply:
Done.